### PR TITLE
Role assignment agent

### PR DESCRIPTION
**Pull request by ZhangT-tech**
## Description

Fix the issue by allowing users input only the task, without need to worry about what specific roles should be assigned to ai user and ai assistant. A new agent called RoleAssignmentAgent has been designed to assign the roles automatically.

## Motivation and Context

Make it more intelligence, without the need for users to think about what should be the most appropriate roles to assign

The issue that is solved can be found here:
https://github.com/camel-ai/camel/issues/129

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [x] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [x] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: camel/__init__.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 import camel.typing
 import camel.utils
 
-__version__ = '0.0.1'
+__version__ = '0.1.0'
```
Comment (Obs01ete) [inline camel/__init__.py:9]: Revert. This is irrelevant to the ticket. Please make another ticket to fix this bug.

File: README.md [hunk 2]
```diff
@@ -27,10 +27,10 @@ Install `CAMEL` from source with conda:
 ```
 # create a conda virtual environment
 conda create --name camel python=3.10
-# actiavte camel conda environment
+# activate camel conda environment
 conda activate camel
 # clone github repo
-git clone https://github.com/lightaime/camel.git
+git clone -b v0.1.0 https://github.com/lightaime/camel.git
```
Comment (Obs01ete) [inline README.md:33]: Please move out-of-scope changes to another MR

File: camel/agents/chat_agent.py [hunk 3]
```diff
@@ -119,6 +119,7 @@ def step(
                 session.
         """
         messages = self.update_messages(input_message)
+
```
Comment (Obs01ete) [inline camel/agents/chat_agent.py:122]: Revert

File: camel/agents/chat_agent.py [hunk 4]
```diff
@@ -127,6 +128,7 @@ def step(
         num_tokens = num_tokens_from_messages(openai_messages, self.model)
 
         if num_tokens < self.model_token_limit:
+
```
Comment (Obs01ete) [inline camel/agents/chat_agent.py:131]: Revert

File: camel/agents/task_agent.py [hunk 5]
```diff
@@ -151,3 +154,75 @@ def step(
 
         sub_tasks_msg = sub_tasks_msgs[0]
         return TextPrompt(sub_tasks_msg.content)
+
+
+class RoleAssignmentAgent(ChatAgent):
```
Comment (Obs01ete) [inline camel/agents/task_agent.py:159]: Move to a separate file role_assignment_agent.py

File: camel/agents/task_agent.py [hunk 6]
```diff
@@ -151,3 +154,75 @@ def step(
...
+
+            ai_user_role_name = names[0].split(":")[1].strip()
+            ai_assistant_role_name = names[1].split(":")[1].strip()
+
+        if ai_user_role_name is None or ai_assistant_role_name is None:
+            raise RuntimeError("Got None or insufficient Role messages. ")
+        if terminated:
+            raise RuntimeError("Role assignment failed.")
+
+        return ai_user_role_name, ai_assistant_role_name
```
Comment (Obs01ete) [inline camel/agents/task_agent.py:228]: Let's make it a dictionary `dict(ai_user_role_name =..., ai_assistant_role_name=...)`

File: examples/ai_society/role_playing.py [hunk 7]
```diff
@@ -1,14 +1,23 @@
...
 
 
 def main() -> None:
-    task_prompt = "Develop a trading bot for the stock market"
+    task_prompt = "Develop bot for company evaluation"
+
+    role_assignment_agent = RoleAssignmentAgent(model=ModelType.GPT_3_5_TURBO)
+
+    roles = role_assignment_agent.step(task_prompt)
+    ai_user_role = roles[0]
```
Comment (Obs01ete) [inline examples/ai_society/role_playing.py:15]: Rework to the dictionary to avoid magic numbers 0, 1

File: examples/ai_society/role_playing.py [hunk 8]
```diff
@@ -1,14 +1,23 @@
...
+from camel.agents.task_agent import RoleAssignmentAgent
+from camel.typing import ModelType
 from camel.utils import print_text_animated
 
 
 def main() -> None:
-    task_prompt = "Develop a trading bot for the stock market"
+    task_prompt = "Develop bot for company evaluation"
+
+    role_assignment_agent = RoleAssignmentAgent(model=ModelType.GPT_3_5_TURBO)
```
Comment (Obs01ete) [inline examples/ai_society/role_playing.py:12]: This must be a separate example role_playing_with_role_assignement.py

File: camel/agents/task_agent.py [hunk 9]
```diff
@@ -80,9 +80,12 @@ def step(
             self.task_specify_prompt = (self.task_specify_prompt.format(
                 **meta_dict))
 
+        print("ready to go")
```
Comment (Obs01ete) [inline camel/agents/task_agent.py:83]: Remove prints

File: camel/agents/task_agent.py [hunk 10]
```diff
@@ -151,3 +154,75 @@ def step(
...
+            Returns:
+                Tuple[TextPrompt, Textprompt]: The role names generated by the
+                agent for
+                the AI user and AI assistant, respectively.
+        """
+
+        self.reset()
+
+        task_msg = AssistantChatMessage(
+            role_name="Role Assigner",
```
Comment (Obs01ete) [inline camel/agents/task_agent.py:208]: AssistantChatMessage is long gone. Where did you branch from?

**Review by Obs01ete — CHANGES_REQUESTED** [review]

**Comment by lightaime** [comment]
Thanks for the PR @ZhangT-tech. But your branch seems to be outdated. You may first need to update your branch to the current master branch.

File: camel/agents/task_agent.py (hunk 10 above)
Comment (ZhangT-tech) [reply camel/agents/task_agent.py]: I modified based on this version of code: git clone -b v0.1.0 https://github.com/camel-ai/camel.git

Comment (Obs01ete) [reply camel/agents/task_agent.py:204]: @lightaime You were right about the possible misinterpretation of this line in the instruction...

**Review by Obs01ete — CHANGES_REQUESTED** [review]
I suggest to checkout master/HEAD, branch from there and re-create this MR from there

**Pull request closed without merging**